rest_api/simple_supply_rest_api/main.py

```diff
@@ -0,0 +1,147 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import argparse
+import asyncio
+import logging
+import sys
+
+from zmq.asyncio import ZMQEventLoop
+
+from sawtooth_sdk.processor.log import init_console_logging
+
+from sawtooth_rest_api.messaging import Connection
+
+from aiohttp import web
+
+from simple_supply_rest_api.route_handler import RouteHandler
+from simple_supply_rest_api.database import Database
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser(
+        description='Starts the Simple Supply REST API')
+
+    parser.add_argument(
+        '-B', '--bind',
+        help='identify host and port for api to run on',
+        default='localhost:8000')
+    parser.add_argument(
+        '-C', '--connect',
+        help='specify URL to connect to a running validator',
+        default='tcp://localhost:4004')
+    parser.add_argument(
+        '-t', '--timeout',
+        help='set time (in seconds) to wait for a validator response',
+        default=500)
+    parser.add_argument(
+        '--db-name',
+        help='The name of the database',
+        default='simple-supply')
+    parser.add_argument(
+        '--db-host',
+        help='The host of the database',
+        default='localhost')
+    parser.add_argument(
+        '--db-port',
+        help='The port of the database',
+        default='5432')
+    parser.add_argument(
+        '--db-user',
+        help='The authorized user of the database',
+        default='sawtooth')
+    parser.add_argument(
+        '--db-password',
+        help="The authorized user's password for database access",
+        default='sawtooth')
+    parser.add_argument(
+        '-v', '--verbose',
+        action='count',
+        default=0,
+        help='enable more verbose output to stderr')
+
+    return parser.parse_args(args)
+
+
+def start_rest_api(host, port, validator_connection, database):
+    loop = asyncio.get_event_loop()
+    validator_connection.open()
+    app = web.Application(loop=loop)
+    app.on_cleanup.append(lambda app: validator_connection.close())
+
+    handler = RouteHandler(loop, validator_connection, database)
+
+    app.router.add_post('/authorization', handler.authorize)
+
+    app.router.add_post('/agents', handler.create_agent)
+    app.router.add_get('/agents', handler.list_agents)
+    app.router.add_get('/agents/{agent_id}', handler.fetch_agent)
+
+    app.router.add_post('/records', handler.create_record)
+    app.router.add_get('/records', handler.list_records)
+    app.router.add_get('/records/{record_id}', handler.fetch_record)
+    app.router.add_patch(
```

Comment:
I think this probably makes more sense as a post - patch implies that you are modifying the record directly.  Since these are (potentially) asynchronous changes, it probably makes sense to be posts

Reply:
Good point - updated the PR